pkg/providers/instance/instance.go

```diff
@@ -533,25 +548,45 @@ func (p *DefaultProvider) launchInstance(
 
 	log.FromContext(ctx).V(1).Info(fmt.Sprintf("Creating virtual machine %s (%s)", resourceName, instanceType.Name))
 	// Uses AZ Client to create a new virtual machine using the vm object we prepared earlier
-	resp, err := p.createVirtualMachine(ctx, vm, resourceName)
+	poller, err := p.createVirtualMachine(ctx, vm, resourceName)
 	if err != nil {
 		azErr := p.handleResponseErrors(ctx, instanceType, zone, capacityType, err)
 		return nil, azErr
 	}
 
-	if p.provisionMode == consts.ProvisionModeBootstrappingClient {
-		err = p.createCSExtension(ctx, resourceName, launchTemplate.CustomScriptsCSE, launchTemplate.IsWindows)
-		if err != nil {
-			// This should fall back to cleanupAzureResources
-			return nil, err
-		}
-	}
+	// Patch the VM object to fill out a few fields that are needed later.
+	// This is a bit of a hack that saves us doing a GET now.
+	// The reason to avoid a GET is that it can fail, and if it does the future above will be lost,
+	// which we don't want.
+	vm.ID = lo.ToPtr(fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/virtualMachines/%s", p.subscriptionID, p.resourceGroup, resourceName))
+	vm.Properties.TimeCreated = lo.ToPtr(time.Now())
 
-	err = p.createAKSIdentifyingExtension(ctx, resourceName)
-	if err != nil {
-		return nil, err
-	}
-	return resp, nil
+	return &VirtualMachinePromise{
+		Wait: func() error {
+			_, err = poller.PollUntilDone(ctx, nil)
```

Comment:
Do we have to wait for this before initiating CSE create?

Reply:
It's what we were doing before and I wanted to keep it the same just from a safety perspective.

Reply:
@charliedmcb may have tried without wait?